### ISSUE TITLE
`mod` lowers to `KnownFn::Rem`, giving `rem` semantics in every compiled tier

### ISSUE THREAD

**Issue by csm** [csm/clojurust#300]
## Summary

`mod` and `rem` are mapped to the same `KnownFn`, so any lowered function computes `rem` semantics where the source said `mod`. For negative operands these differ, and the result is a silent wrong answer that depends on which tier executed the code.

## Detail

`crates/cljrs-ir/src/lower/known.rs:57-58`:

```rust
"rem" => Some(KnownFn::Rem),
"mod" => Some(KnownFn::Rem),   // <- same variant
```

Nothing downstream distinguishes the two names. `lower_known_call` receives the original `name` but never consults it for this case, and every consumer treats the variant as `rem`:

- `crates/cljrs-eval/src/ir_interp.rs:1135` — `KnownFn::Rem => builtin_arith(&args, "rem")`
- `crates/cljrs-compiler/src/codegen.rs:2172` — `KnownFn::Rem => self.rt.rt_rem`
- `crates/cljrs-compiler/src/wasm/emit.rs:1426` — `KnownFn::Rem => ("rt_rem", false)`

Meanwhile the tree-walking tier is correct: `builtin_mod` (`crates/cljrs-builtins/src/builtins.rs:2361`) implements Clojure's sign-of-divisor semantics, including the NaN/infinity edge cases.

## Impact

```clojure
(mod -10 3)
;; => 2   in the tree-walk tier (correct, matches Clojure)
;; => -1  once the enclosing fn is lowered to IR / JIT / AOT
```

No error is raised. The answer changes based on whether the function crossed the warm/JIT threshold, was lowered eagerly, or was AOT-compiled.

## Why the suite doesn't catch it

`cargo test --workspace` is green (1001 passed, 0 failed), and `clojure-test-suite/test/clojure/core_test/mod.cljc:9-15` already asserts the correct values:

```clojure
int? 1  10  3
int? 2  -10 3
int? -1 -10 -3
int? -2 10  -3
```

Since these pass, nothing currently exercises `mod` with a negative operand at a lowered tier. The divergence is latent rather than caught.

## Suggested fix

1. Add a distinct `KnownFn::Mod` variant.
2. Point `"mod"` at it in `known.rs`.
3. Add an `rt_mod` runtime entry point and wire the variant through `ir_interp.rs`, `codegen.rs`, and `wasm/emit.rs`.
4. Check the other match sites that currently list `KnownFn::Rem` in a group — `crates/cljrs-ir/src/lower/anf.rs:2241`, `crates/cljrs-ir/src/lower/optimize.rs:514`, `crates/cljrs-ir/src/lib.rs:1019`, `crates/cljrs-compiler/src/escape.rs` — and classify `Mod` the same way (both are pure, total-except-div-by-zero binary numeric ops).
5. Add a regression test that pins the interpreter and IR tiers to the same answer for negative operands, so this class of divergence is caught rather than depending on tier.

Note that `crates/cljrs-ir/src/lower/anf.rs:2469` and `:2482` emit `KnownFn::Rem` directly as part of an inline expansion — those are genuine `rem` uses and should keep using `Rem`.
